starter/src/main/jni/starter.c

```diff
@@ -0,0 +1,110 @@
+#include <jni.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/poll.h>
+#include <sys/un.h>
+#include <errno.h>
+#include <string.h>
+#include <fcntl.h>
+
+#define unused __attribute__((__unused__))
+
+#define PREFIX "/data/data/com.termux/files/usr"
+#define XDG_RUNTIME_DIR PREFIX "/tmp"
```

Comment:
Also.. please don't hard-code `XDG_RUNTIME_DIR` We currently use user-defined `XDG_RUNTIME_DIR`.
You can see `app/src/main/res/xml/preferences.xml`

Reply:
Ok, I'll fix it. I need to check how to read it inside starter.apk.

Reply:
Ok. The situation is simple: termux-x11 app does not handle creating sockets anymore, it is handled in starter.apk. I can put there checking if XDG_RUNTIME_DIR is set and if not export default value.